src/tasks/MonoTargetsTasks/MarshalingPInvokeScanner/MarshalingPInvokeScanner.cs

```diff
@@ -0,0 +1,333 @@
+    public Compatibility GetByReferenceType(Compatibility elementType) => Compatibility.Compatible;
+    public Compatibility GetFunctionPointerType(MethodSignature<Compatibility> signature) => Compatibility.Compatible;
+    public Compatibility GetGenericInstantiation(Compatibility gennricType, ImmutableArray<Compatibility> typeArguments) => Compatibility.Incompatible;
+    public Compatibility GetGenericMethodParameter(object genericContext, int index) => Compatibility.Incompatible;
+    public Compatibility GetGenericTypeParameter(object genericContext, int index) => Compatibility.Incompatible;
+    public Compatibility GetModifiedType(Compatibility modifier, Compatibility unmodifiedType, bool isRequired) => Compatibility.Incompatible;
+    public Compatibility GetPinnedType(Compatibility elementType) => Compatibility.Compatible; // TODO: really?
+    public Compatibility GetPointerType(Compatibility elementType) => Compatibility.Compatible;
+    public Compatibility GetPrimitiveType(PrimitiveTypeCode typeCode)
+    {
+        return typeCode switch
+        {
+           PrimitiveTypeCode.Object => Compatibility.Incompatible,
+           PrimitiveTypeCode.String => Compatibility.Incompatible,
+           PrimitiveTypeCode.TypedReference => Compatibility.Incompatible, // TODO: really?
+           _ => Compatibility.Compatible
+        };
+    }
+
+    public Compatibility GetSZArrayType(Compatibility elementType) => Compatibility.Incompatible;
+
+    public Compatibility GetTypeFromDefinition(MetadataReader reader, TypeDefinitionHandle handle, byte rawTypeKind)
+    {
+        TypeDefinition typeDef = reader.GetTypeDefinition(handle);
+        if (reader.GetString(typeDef.Name) == "Enum")
+            return Compatibility.Compatible;
+
+        try
+        {
+            EntityHandle baseTypeHandle = typeDef.BaseType;
+            if(baseTypeHandle.Kind == HandleKind.TypeReference)
+            {
+                TypeReference baseType = reader.GetTypeReference((TypeReferenceHandle)baseTypeHandle);
+                if (reader.GetString(baseType.Name) == "Enum")
+                    return Compatibility.Compatible;
+            }
+            else if(baseTypeHandle.Kind == HandleKind.TypeSpecification)
+            {
+                TypeSpecification specInner = reader.GetTypeSpecification((TypeSpecificationHandle)baseTypeHandle);
+                return specInner.DecodeSignature<Compatibility, object>(this, new object());
+            }
+            else if(baseTypeHandle.Kind == HandleKind.TypeDefinition)
+            {
+                TypeDefinitionHandle handleInner = (TypeDefinitionHandle)baseTypeHandle;
+                if(handle != handleInner)
+                    return GetTypeFromDefinition(reader, handleInner, rawTypeKind);
+            }
+        }
+        catch(BadImageFormatException)
+        {
+            return Compatibility.Incompatible;
```

Comment:
Would it be useful to log a message about the assembly name that couldn't be processed here?